lib/puppet/configurer.rb

```diff
@@ -224,9 +224,10 @@ def run(options = {})
         if options[:catalog].nil? && do_failover
           server, port = find_functional_server
           if server.nil?
-            raise Puppet::Error, _("Could not select a functional puppet master from server_list: '%{server_list}'") % { server_list: Puppet.settings.value(:server_list, Puppet[:environment].to_sym, true) }
+            raise Puppet::Error, _("Could not select a functional puppet server from server_list: '%{server_list}'") % { server_list: Puppet.settings.value(:server_list, Puppet[:environment].to_sym, true) }
           else
             report.master_used = "#{server}:#{port}"
+            report.server_used = "#{server}:#{port}"
```

Comment:
Should we remove the `Report#master_used` method and have `server_used` populate both entries in the report? Or did you want to save the removal of that method to Puppet 7?

Reply:
Yes ... I don't know why I have not done like this in the first time, sorry